### ISSUE TITLE
Calva overwriting highlighting of the selected theme

### ISSUE THREAD

**Issue by avillega** [BetterThanTomorrow/calva#1090]
Hi I've been using Calva for a couple of days and love the experience so far. The main difficulty has been around highlighting. I am user of minimal themes like Alabaster and it looks like Calva does not go well with them. This is a screenshot of the editor when calva is enabled as an extension (the REPL is disconnected but this happens when is connected aswell)
![image](https://user-images.githubusercontent.com/11149741/113174205-fef32380-91fe-11eb-8655-178eca229db3.png)
And this is a picture of the same editor but with Calva disabled
![image](https://user-images.githubusercontent.com/11149741/113174373-2813b400-91ff-11eb-8f44-7a5edb663cf8.png)

This are my specific configurations
```
    "calva.highlight.rainbowIndentGuides": false,
    "calva.highlight.enableBracketColors": false,
    "calva.highlight.highlightActiveIndent": false,
```
if I keep the defaults, the behaviour is the same
![image](https://user-images.githubusercontent.com/11149741/113174850-a708ec80-91ff-11eb-8c55-c26b694787c2.png)

Thanks for this great extension and all the hard work you put into it!

**Comment by PEZ** [comment]
Hello, thanks for reaching out. I need help to see the issue.

To me the first and second image look the same, except for the keywords. Keywords are lexed as `constant` by the Calva syntax grammar. 

The third image looks different though, so ”the behaviour is the same” makes no sense to me. 😄

**Comment by avillega** [comment]
HI @PEZ thanks for the quick response. Yes, the main difference is the keywords, which is basically the only thing that the minimal theme highlights. The problem is that Calva is overwriting the way the theme highlights.

**Comment by avillega** [comment]
I added this TextMate rule to my settings and was the only one that make it work
```
            "textMateRules": [
                {
                    "scope": "variable.other",
                    "settings": {
                        "foreground": "#FF0000"
                    }
                }
            ]
        }
```
I believe something may be going on while selecting the color to use, because the theme has a color for constants.  I have so time to tomorrow to probably take a look into this, could be a fun first contribution to Calva!

**Comment by PEZ** [comment]
VS Code has a command **Inspect Editor tokens and scopes** that can shed some light on it.

**Comment by avillega** [comment]
I found what I believe is the culprit. in this file [src/calva-fmt/atom-language-clojure/grammars/clojure.cson](https://github.com/BetterThanTomorrow/calva/blob/published/src/calva-fmt/atom-language-clojure/grammars/clojure.cson ) Yo can see that keywords are being parsed as follows
```
  'keyword':
    'match': '(?<=^|\\s|\\(|\\[|\\{):[\\w\\#\\.\\-\\_\\:\\+\\=\\>\\<\\/\\!\\?\\*]+(?=(\\s|\\)|\\]|\\}|\\,))'
    'name': 'variable.other.constant.clojure'
```
As you can see it is being parse as a `variable.other.constant.clojure` I would expect it to be a `constant.keyword.clojure` instead. The `constant.keyword.clojure` matches with how other tokens are being parsed and also with what I've seen the themes use. @PEZ What do you think about this change? Is there any reason to use `variable.other.constant.clojure` instead `constant.keyword.clojure`. If you are ok with the change I could create a PR with it.

**Comment by PEZ** [comment]
That's what it used to be. I don't fully remember why we had to change it, but it had to do with being able to style the kwywords. See:

https://github.com/BetterThanTomorrow/calva/commit/28af917a3575353be5aea6829eb48004565df122

**Comment by avillega** [comment]
Closing this since it looks there is a good reason to parse keywords as it currently is.

**Comment by PEZ** [comment]
Not sure we should close this just yet. 😄 The choice of `variable.other.constant.clojure` was a bit arbitrary. I experimented a while and picked one that played best with the Themes I tested with. I think it could be worth experimenting some more to se if something can be done to make it play better with that theme you are using, while also keeping the distinct styling of keywords in the more common themes.
